plugin/cql/src/main/java/org/opencds/cqf/ruler/cql/KnowledgeArtifactProcessor.java

```diff
@@ -376,20 +397,37 @@ public Bundle createReleaseBundle(IdType idType, String version, CodeType versio
 			// and child artifact components recursively
 			// as root artifact dependencies
 			for(RelatedArtifact component : components){
-				MetadataResource resource = checkIfReferenceInList(component, resourcesToUpdate)
-					.orElseGet(() -> getLatestActiveVersionOfReference(component.getResource(), fhirDal, artifact.getUrl()));
-				String reference = String.format("%s|%s", resource.getUrl(), resource.getVersion());
-				component.setResource(reference);
+				MetadataResource resource;
+				if(KnowledgeArtifactAdapter.checkIfRelatedArtifactIsOwned(component)){
+					resource = checkIfReferenceInList(component, resourcesToUpdate)
+					.orElseGet(() -> {
+						try {
+							return getLatestActiveVersionOfReference(component.getResource(), fhirDal, artifact.getUrl());
+						} catch (ResourceNotFoundException e) {
+							throw new PreconditionFailedException(
+								String.format("Resource with URL '%s' is an owned component of resource '%s', but no active version of that resource was found on this server, nor is there a draft version which can be updated.",
+									component.getResource(),
+									artifact.getUrl()));
+						}
+					});
+					String reference = String.format("%s|%s", resource.getUrl(), resource.getVersion());
+					component.setResource(reference);
```

Comment:
It's currently only attempting to update references to `owned` components, should it attempt to get the latest versions of components which aren't `owned`?